pkg/storage/azureblob.go

```
@@ -14,45 +18,39 @@ type AzureBlobOpts struct {
 	Container   string
 }
 
-// AzureBlob is used to upload files to azure blob storage
-type AzureBlob struct {
-	Container *azure.Container
+// AzureContainer is used to create containers and upload files to Azure blob storage
+type AzureContainer struct {
+	Container *azblob.ContainerURL
 }
 
 // NewAzureBlob creates a new AzureBlob
-func NewAzureBlob(azureOpts *AzureBlobOpts) (*AzureBlob, error) {
+func NewAzureBlob(azureOpts *AzureBlobOpts) (*AzureContainer, error) {
 	if azureOpts == nil {
 		return nil, errors.New("missing azure options")
 	}
 
-	client, err := azure.NewBasicClient(azureOpts.AccountName, azureOpts.AccountKey)
+	credential, err := azblob.NewSharedKeyCredential(azureOpts.AccountName, azureOpts.AccountKey)
 	if err != nil {
 		panic(err)
 	}
 
-	blobClient := client.GetBlobService()
-	containerRef := blobClient.GetContainerReference(azureOpts.Container)
-	containerExists, err := containerRef.Exists()
-	if err != nil {
-		return nil, err
-	}
-	if !containerExists {
-		return nil, fmt.Errorf("azure blob storage container %s does not exist", azureOpts.Container)
-	}
+	pipeline := azblob.NewPipeline(credential, azblob.PipelineOptions{})
+	URL, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s", azureOpts.AccountName, azureOpts.Container))
```

Let's not ignore the error here (second return parameter).